app/assets/javascripts/models/data/collection.js.coffee

```diff
@@ -344,15 +345,39 @@ ns.Collection = do (ko
       @_loadDatasource()
       @granuleDatasource()?.updateFromCollectionData?(jsonObj)
 
-
       if @granuleDatasourceName() && @granuleDatasourceName() != 'cmr'
         @has_granules = @canFocus()
 
+    _computeHasSpatial: ->
+      if @associations()?['services']?.length > 0
+        for serviceId in @associations()['services'] when !@has_spatial()
+          @_getServices(serviceId)
+
     _setObservable: (prop, jsonObj) =>
       this[prop] ?= ko.observable(undefined)
       this[prop](jsonObj[prop] ? this[prop]())
 
+    _getServices: (id)->
+      ajax
+        dataType: 'json'
+        url: "/services/#{id}"
+        success: (data) =>
+          if data['ServiceOptions']?['SubsetType'].length > 0
+            for subsetType in data['ServiceOptions']['SubsetType']
+              if subsetType == 'Spatial'
+                @has_spatial(true)
```

Comment:
This is a placeholder until CMR-4582 is complete, which is currently in progress. It's called has_spatial because its looking through services to see if *any* services have the `Spatial` value included in SubsetType. This method will match the key that is being provided by CMR when the ticket above is complete.